sycl/test-e2e/lit.cfg.py

```diff
@@ -821,6 +826,70 @@ def remove_level_zero_suffix(devices):
 if config.test_mode != "build-only":
     config.sycl_build_targets = set()
 
+# A device filter such as level_zero:gpu can have multiple devices under it and
+# the order is not guaranteed. The aspects enabled are also restricted to what
+# is supported on all devices under the label. It is possible for level_zero:gpu
+# and level_zero:0 to select different devices on different machines with the
+# same hardware. It is not currently possible to pass the device architecture to
+# ONEAPI_DEVICE_SELECTOR. Instead, if "BACKEND:arch-DEVICE_ARCH" is provided to
+# "sycl_devices", giving the desired device architecture, select a device that
+# matches that architecture using the backend:device-num device selection
+# scheme.
+filtered_sycl_devices = []
+for sycl_device in remove_level_zero_suffix(config.sycl_devices):
+
+    backend, device_arch  = sycl_device.split(":", 1)
+
+    if ("arch-" in device_arch):
+        env = copy.copy(llvm_config.config.environment)
+
+        # Find all available devices under the backend
+        env["ONEAPI_DEVICE_SELECTOR"] = backend + ":*"
+
+        # When using the ONEAPI_DEVICE_SELECTOR environment variable, sycl-ls
+        # prints warnings that might derail a user thinking something is wrong
+        # with their test run. It's just us filtering here, so silence them unless
+        # we get an exit status.
```

Comment:
Please outline to some helper instead of copy-pasting. While doing this please take a look `with test_env():`, it might be beneficial/necessary here and in the pre-existing `sycl-ls` invocation.

Reply:
Added helper function and added `with text_env()` good idea to help prevent any environment weirdness.